pkgs/gnolang/preprocess.go

```diff
@@ -2231,6 +2243,15 @@ func lastDecl(ns []Node) Decl {
 	return nil
 }
 
+func lastSwitch(ns []Node) *SwitchStmt {
+	for i := len(ns) - 1; 0 <= i; i-- {
+		if d, ok := ns[i].(*SwitchStmt); ok {
+			return d
+		}
+	}
+	return nil
+}
```

Comment:
I was tempted to create a generic function like

```go
func lastType[T any](ns []Node) T
```

which could replace `lastSwitch` and the previous `lastDecl` but I'm not sure if this acceptable in this context, because of using generics can degrade performances maybe.

Reply:
At this point, I would rather not use generics here either. We should consider if we want to start utilizing generics more widely throughout the rest of the project.

Reply:
got it :+1: